Description: install linux .desktop and hicolor icons
This patch is hardcoded for linux, do not upstream.
Author: Feignint <37488595+FeignInt@users.noreply.github.com>
Last-Update: 2020-07-13
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/configure.ac
+++ b/configure.ac
@@ -661,5 +661,7 @@ src/platform/Makefile
 src/platform/visualc/Makefile
 include/Makefile
 docs/Makefile
+contrib/Makefile
+contrib/linux/Makefile
 ])
 AC_OUTPUT
--- /dev/null
+++ b/contrib/linux/Makefile.am
@@ -0,0 +1,12 @@
+desktopdir = $(datadir)/applications
+desktop_DATA = dosbox-staging.desktop
+iconsdir = $(datadir)/icons
+nobase_dist_icons_DATA = hicolor/*/*/dosbox-staging.*
+
+hicolor/%:
+	make -C $(top_srcdir)/contrib/icons hicolor
+	ln -s $(top_srcdir)/contrib/icons/hicolor .
+
+clean-am:
+	rm hicolor
+	make -C $(top_srcdir)/contrib/icons clean
--- /dev/null
+++ b/contrib/Makefile.am
@@ -0,0 +1 @@
+SUBDIRS = linux
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 # Main Makefile for DOSBox
 
 EXTRA_DIST = autogen.sh
-SUBDIRS = src include docs
+SUBDIRS = src include docs contrib
 
 # Per the auto-tools clean guidelines:
 # https://www.gnu.org/savannah-checkouts/gnu/automake/manual/html_node/Clean.html#Clean
